mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -89,6 +91,7 @@ where
                     }
                     buffer.extend_from_slice(&body);
 
+                    // Ensure the flow-control window is large enough.
                     result.as_mut().increment_read_window(self.block_size as usize);
```

Comment:
This seems unnecessary now, doesn't it?

Reply:
Happy to keep it for now and review when we optimize for the single chunk case (see TODO above).

Reply:
We'll need to account for a case when the block object is larger than `block_size` for some reason. If we've just removed this line the `read` may freeze. If we've kept it as it is now MP may attempt to read an unbounded amount of data to RAM. 

This requires a bit more thinking, so I agree that it's better to address in the following PR.